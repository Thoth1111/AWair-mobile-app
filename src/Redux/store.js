import { configureStore } from '@reduxjs/toolkit';
import thunkMiddleware from 'redux-thunk';
import { homeReducer } from './home/homeSlice';

const store = configureStore({
  reducer: {
    home: homeReducer,
    middleware: [thunkMiddleware],
  },
});

export default store;
