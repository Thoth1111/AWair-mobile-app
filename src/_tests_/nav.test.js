import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Navbar from '../components/navbar';

describe('Navbar', () => {
  it('should render the component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
