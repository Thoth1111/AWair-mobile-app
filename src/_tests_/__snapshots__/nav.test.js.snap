// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
<nav>
  <h1>
    AW
    <span
      id="air-txt"
    >
      ~air
    </span>
  </h1>
  <span
    id="description-txt"
  >
    Global Real-time Air Quality index.
  </span>
  <form
    className="form-section"
    onSubmit={[Function]}
  >
    <label
      htmlFor="cityname"
    >
      <input
        id="cityname"
        name="cityname"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Enter City Name"
        type="text"
        value=""
      />
      <button
        id="sub-btn"
        type="submit"
      >
        <svg
          className="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
          <path
            d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
          />
        </svg>
      </button>
    </label>
  </form>
</nav>
`;
